inference-engine/src/low_precision_transformations/src/transparent_base_transformation.cpp

```diff
@@ -29,13 +31,20 @@ void TransparentBaseTransformation::transform(TransformationContext& context, CN
             CNNNetworkHelper::setOutDataPrecision(layer, getPrecisionBeforeParentDequantizationScaleShift(layer));
         }
 
-        const Blob::Ptr weights_blob = CNNNetworkHelper::getBlob(scaleShift, "weights");
-        auto weights = CNNNetworkHelper::getFloatData(weights_blob);
-        const std::vector<float> scales = std::vector<float>(weights.get(), weights.get() + weights_blob->size());
+        std::vector<float> scales;
+        std::vector<float> shifts;
+        fillFromDequantizationLayer(*scaleShift, scales, shifts);
 
-        const Blob::Ptr biases_blob = CNNNetworkHelper::getBlob(scaleShift, "biases");
-        auto biases = CNNNetworkHelper::getFloatData(biases_blob);
-        const std::vector<float> shifts = std::vector<float>(biases.get(), biases.get() + biases_blob->size());
+        const size_t inputChannels = CNNNetworkHelper::getInputChannelsCount(layer);
+        const size_t outputChannels = CNNNetworkHelper::getOutputChannelsCount(layer);
+        if (inputChannels != outputChannels) {
+            if (!DequantizationDetails::isPerTensor(scales, shifts)) {
+                THROW_IE_LPT_EXCEPTION(layer) << "layer input and output channels count values are different for by channel quantization";
```

Comment:
Why exception? Max pooling should be handled by transparent transformation and it easily can support per-channel granularity

Reply:
Exception happens only if input and output channels values are different